### PR TITLE
retry for arxiv PDF download

### PR DESCRIPTION
**Pull request by monaawi**
Caught request exception in arxiv pdf download and threw them as DownloadError. Added backoff for IO error.
INSPIR-1478

### CODE REVIEW

**Comment by monaawi** [comment]
ci test please

File: inspirehep/modules/workflows/tasks/arxiv.py [hunk 1]
```diff
@@ -73,10 +73,12 @@ def populate_arxiv_document(obj, eng):
...
-
-        if NO_PDF_ON_ARXIV in requests.get(url).content:
-            obj.log.info('No PDF is available for %s', arxiv_id)
-            return
+        try:
+            if NO_PDF_ON_ARXIV in requests.get(url).content:
+                obj.log.info('No PDF is available for %s', arxiv_id)
+                return
+        except requests.exceptions.RequestException:
+            raise DownloadError()
```
Comment (drjova) [inline inspirehep/modules/workflows/tasks/arxiv.py]: maybe add information about the url?

File: tests/unit/workflows/test_workflows_tasks_arxiv.py [hunk 2]
```diff
@@ -506,6 +543,51 @@ def test_arxiv_plot_extract_handles_duplicate_plot_names(mock_os, tmpdir):
...
+    assert validate(data['arxiv_eprints'], subschema) is None
+
+    obj = MockObj(data, extra_data, files=files)
+    eng = MockEng()
+
+    temporary_dir = tmpdir.mkdir('plots')
+    mock_os.path.abspath.return_value = str(temporary_dir)
+
+    with pytest.raises(IOError):
+        with patch('__builtin__.open') as mock_open:
```
Comment (drjova) [inline tests/unit/workflows/test_workflows_tasks_arxiv.py]: better here to patch ```open``` for the module you are testing

File: tests/unit/workflows/test_workflows_tasks_arxiv.py [hunk 3]
```diff
@@ -287,6 +290,40 @@ def test_populate_arxiv_document_retries_on_error():
         assert expected_documents == documents
 
 
+def side_effect_requests_get(url, params=None, **kwargs):
+    raise requests.exceptions.ConnectionError()
+
+
+@patch('requests.get')
```
Comment (drjova) [inline tests/unit/workflows/test_workflows_tasks_arxiv.py]: more specific namespace will be better for the patch

**Comment by monaawi** [comment]
ci test please unit

**Comment by monaawi** [comment]
ci test please

**Comment by monaawi** [comment]
ci merge please

**Pull request merged**